lib/active_model/serializer/array_serializer.rb

```diff
@@ -5,7 +5,7 @@ class ArraySerializer
       include Enumerable
       delegate :each, to: :@objects
 
-      attr_reader :root, :meta, :meta_key
+      attr_reader :root, :meta, :meta_key, :options, :resource
```

Comment:
re: adding `options` and `resource` as public methods on the ArraySerializer so that the json-api adapter may use them

So, in theory, the serializer and array serializer just represent transforming the resource into a serializable hash (of attributes), and the adapter presents that hash according to the type of adapter it is, e.g. json-api adapter.  That's how I'm thinking of it, at least.  If so, then per https://github.com/rails-api/active_model_serializers/blob/03ddeeeec1e4eedafeb68eeccd59e8c3826b859d/lib/active_model/serializable_resource.rb#L5 any options passed to the `SerializableResource` that pertain to the presentation of the resource should go to the adapter, which I think is a good match for `pagination`. That said, `meta` goes to the serializer now, though it is used by the adapter... so, I guess we'll see what makes more sense.  cc @joaomdmoura @kurko